src/prover/proving_service.rs

```diff
@@ -1,16 +1,16 @@
-use super::CircuitType;
+use super::ProofType;
 use async_trait::async_trait;
 
 #[async_trait]
 pub trait ProvingService {
     fn is_local(&self) -> bool;
     async fn get_vks(&self, req: GetVkRequest) -> GetVkResponse;
-    async fn prove(&self, req: ProveRequest) -> ProveResponse;
-    async fn query_task(&self, req: QueryTaskRequest) -> QueryTaskResponse;
+    async fn prove(&mut self, req: ProveRequest) -> ProveResponse;
+    async fn query_task(&mut self, req: QueryTaskRequest) -> QueryTaskResponse;
```

Comment:
A query action which would change the service seems a little confusion, do we really need this entry not to be idempotent?

Reply:
Our prover implementations unfortunately do require mutable access right now. It could be worked around, but it just gets ugly really quick.